x-pack/plugins/encrypted_saved_objects/server/lib/audit_logger.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export class EncryptedSavedObjectsAuditLogger {
+  constructor(private readonly enabled: boolean, private readonly auditLogger: any) {}
+
+  public encryptAttributeFailure(attributeName: string, type: string, id?: string) {
+    if (!this.enabled) {
+      return;
+    }
+
+    this.auditLogger.log(
+      'encrypt_failure',
+      `Failed to encrypt attribute "${attributeName}" for saved object "${type}:${id}".`,
+      { id, type, attributeName }
+    );
+  }
+
+  public decryptAttributeFailure(attributeName: string, type: string, id: string) {
+    if (!this.enabled) {
+      return;
+    }
+
+    this.auditLogger.log(
+      'decrypt_failure',
+      `Failed to decrypt attribute "${attributeName}" for saved object "${type}:${id}".`,
+      { id, type, attributeName }
+    );
+  }
+
+  public encryptAttributesSuccess(attributeNames: ReadonlySet<string>, type: string, id?: string) {
+    if (!this.enabled) {
+      return;
+    }
+
+    this.auditLogger.log(
+      'encrypt_success',
+      `Successfully encrypted attributes "${Array.from(
+        attributeNames
+      )}" for saved object "${type}:${id}".`,
+      { id, type, attributeNames }
```

Comment:
__note:__ I'll add `Array.from` here as well, JSON won't be able to "pretty-print" `Set`.